nodeinfo/manager.go

```diff
@@ -60,73 +59,74 @@ type (
 	}
 )
 
-var nodeDelegateHeightGauge = prometheus.NewGaugeVec(
+const _nodeMapSizeLimit = 1000
+
+var _nodeInfoHeightGauge = prometheus.NewGaugeVec(
 	prometheus.GaugeOpts{
-		Name: "iotex_node_delegate_height_gauge",
-		Help: "delegate height",
+		Name: "iotex_node_info_height_gauge",
+		Help: "height info of node",
 	},
 	[]string{"address", "version"},
 )
 
 func init() {
-	prometheus.MustRegister(nodeDelegateHeightGauge)
+	prometheus.MustRegister(_nodeInfoHeightGauge)
 }
 
-// NewDelegateManager new delegate manager
-func NewDelegateManager(cfg *Config, t transmitter, h chain, privKey crypto.PrivateKey) *DelegateManager {
-	dm := &DelegateManager{
-		cfg:         *cfg,
-		nodeMap:     make(map[string]Info),
+// NewInfoManager new info manager
+func NewInfoManager(cfg *Config, t transmitter, h chain, privKey crypto.PrivateKey) *InfoManager {
+	dm := &InfoManager{
+		nodeMap:     lru.New(_nodeMapSizeLimit),
 		transmitter: t,
 		chain:       h,
 		privKey:     privKey,
-		myVersion:   version.PackageVersion,
-		myAddress:   privKey.PublicKey().Address().String(),
+		version:     version.PackageVersion,
+		address:     privKey.PublicKey().Address().String(),
 	}
 
 	dm.broadcastTask = routine.NewRecurringTask(func() {
 		// delegates and nodes who are turned on will broadcast
-		if !dm.isDelegate() && dm.cfg.OnlyDelegateBroadcast {
-			log.L().Debug("delegate manager general node disabled node info broadcast")
+		if !cfg.EnableBroadcastNodeInfo && !dm.isDelegate() {
```

Comment:
what if `EnableBroadcastNodeInfo` is false and `isDelegate()` is true

Reply:
It will broadcast in this case. That is to say delegate will always broadcast.